src/dynsight/_internal/analysis/rdf.py

```diff
@@ -0,0 +1,179 @@
+        if topology_file is None:
+            self.pipeline = import_file(self.trajectory, columns=xyz_cols)
+        else:
+            self.pipeline = import_file(self.trajectory[0])
+            timeframes = LoadTrajectoryModifier()
+            timeframes.source.load(str(self.trajectory[1]))
+            self.pipeline.modifiers.append(timeframes)
+
+    def compute_rdf(
+        self,
+        cutoff: float,
+        bins: int = 200,
+        remove_atoms: str | None = None,
+        step: int = 1,
+    ) -> None:
+        """Compute the Radial Distribution Function (RDF).
+
+        Args:
+            cutoff (float):
+                Maximum distance for calculating the RDF.
+
+            bins (int):
+                Number of bins for the RDF calculation, Default is 200.
+
+            remove_atoms (str | None, optional):
+                Selection OVITO expression for atoms
+                to be removed (e.g., 'ParticleType == 1').
+
+            step (int, optional):
+                Frequency of time sampling for averaging
+                the RDF. Default is 1.
+
+        Important:
+            If you're using the `remove_atoms` argument to remove certain
+            particles from the RDF calculation, you need to use the
+            OVITO's expression syntax for selecting particles.
+
+            Example expressions:
+                - `"ParticleType == 1"`: Selects all particles of type 1.
+
+                - `"Position.X > 0"`: Selects particles with an x-coordinate
+                    greater than zero.
+
+                - `"ParticleIdentifier % 2 == 0"`: Selects particles with even
+                    particle identifiers.
+
+            Be sure to have this type of information stored in your
+            simulation file before using this command.
+
+            More information on OVITO's expression syntax can be found here:
+            https://www.ovito.org/manual/reference/pipelines/modifiers/expression_select.html
+        """
+        # Delete selected atoms (OVITO commands)
+        if remove_atoms is not None:
+            selection_modifier = ExpressionSelectionModifier(
+                expression=remove_atoms
+            )
+            self.pipeline.modifiers.append(selection_modifier)
+            self.pipeline.modifiers.append(DeleteSelectedModifier())
+
+        # Compute RDF
+        coord_modifier = CoordinationAnalysisModifier(
+            cutoff=cutoff, number_of_bins=bins
+        )
+        self.pipeline.modifiers.append(coord_modifier)
+
+        # Time averaging
+        averaging_modifier = TimeAveragingModifier(
+            operate_on="table:coordination-rdf", sampling_frequency=step
+        )
+        self.pipeline.modifiers.append(averaging_modifier)
+
+        # Compile the pipeline
+        data = self.pipeline.compute()
+        total_rdf = data.tables["coordination-rdf[average]"].xy()
+        self.pair_distances = total_rdf[:, 0]
+        self.rdf = total_rdf[:, 1]
```

Comment:
My concern here is that this variable is not safe.

A user can go `RDF().rdf = 1` and it would change the RDF of the system and they would not know.

I think we should impose immutable classes for analysis classes to ensure this does not happen. The simplest way to do that now is:

- [ ] minimise the number of attributes (remove unnecessary `self.variable`)
- [ ] use private attributes: `self._rdf = total_rdf[:,1]`, so the user cannot access it
- [ ] provide "getters": `RDF().get_rdf()`, which just returns `self._rdf`.